### ISSUE TITLE
Create Backend 

### ISSUE THREAD

**Issue by akhilam512** [akhilam512/AsteroidTracker#2]
The backend is planned to work in this manner - 

  - Send HTTP requests to NASA Open API (https://api.nasa.gov/) for asteroid data
  - Collect data
  - Parse it and show it in a table 

A backend with these minimum features in the initial round is needed.

**Comment by rpartha** [comment]
I'd like to handle this task on my own. Also, by table you mean database table or just log this information in table format in GUI?

**Comment by Freakston** [comment]
Yeah feel free to work on it 👍

Okay well if you have a look at the Readme file you would see that we use  NASA API to get the data on asteroids and then we make a table with the details. Yes we are trying to log this information in a table format(GUI)  such that each asteroid would have a separate page of its own information .

**Comment by Freakston** [comment]
What we are trying to achieve in this project is an application which tells you how many asteroids pass by the earth on each day and when you click on one of the asteroids it should open a new window and display information about it (like the diameter,speed,etc) which is readily available form the API .

**Comment by alrobleslopez** [comment]
Hi guys. I would like to collaborate as well in this project. Is it possible? If possible, we could coordinate @rpartha

**Comment by rpartha** [comment]
@alrobleslopez maybe you can expand on this -- I wanted to cover the basics, and then you can expand on it?

**Comment by rpartha** [comment]
I was actually planning on plotting some of the data maybe using JFreeChart. If you want, you can do that.

**Comment by rpartha** [comment]
I'll probably begin work tonight, so if I'll lyk when you can start doing stuff.

**Comment by akhilam512** [comment]
> I was actually planning on plotting some of the data maybe using JFreeChart. If you want, you can do that.

That sounds like a great idea @rpartha, you could maybe get started with it and send a WIP. What do you think @rpartha, @alrobleslopez ?

**Comment by alrobleslopez** [comment]
When I read "Create backend" I was thinking about create a SpringBoot/whatever framework API, in order to be consumed from another visual component. Is that feasible @akhilam512 @rpartha ?

**Comment by Freakston** [comment]
Hey @alrobleslopez Well as of now our plan for the back-end was to use the API to get the data and then parse it. If you have a look at the API it sends you the formatted data. Then we need to make table entries and other visual components using the parsed data. 

If you have a better idea/implementation of the back-end then you are free to proceed. You could state your idea on this thread too, this way we will be having a better view of your plan. I will shortly be adding the code for the post request and parsing.

**Comment by akhilam512** [comment]
> When I read "Create backend" I was thinking about create a SpringBoot/whatever framework API, in order to be consumed from another visual component. Is that feasible @akhilam512 @rpartha ?

That makes sense, yes, we would need some sort of a framework API on top of which we could do the requests and I think SpringBoot would work just fine.  Isn't that right @Freakston ? The requests can go from the SpringBoot backend.

**Comment by Freakston** [comment]
Yes @akhilam512 . I think it would be great if @alrobleslopez can use springboot and implement the back-end.

**Comment by alrobleslopez** [comment]
OK guys, so if you are agree, I'll mount an SpringBoot backend that consumes NASA API and serves the information as required by "the frontend/GUI". In the other hand, do you have a chat/slack/whatever to ease communication between us? Thanks !

**Comment by Freakston** [comment]
@alrobleslopez Feel free to ping me on Telegram. Telgram id - Freakston.

**Comment by rpartha** [comment]
Telegram - m4st3r0w1

**Comment by akhilam512** [comment]
You may join the Telegram group here - 

 https://t.me/AsteroidTracker

**Comment by nurettn** [comment]
The backend side is done with PR#7. Can you merge please?
